bundles/umljava/tools.vitruv.applications.umljava.uml2java/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions

```diff
@@ -30,37 +30,71 @@ execute actions in Java
 
 reaction UmlClassCreated {
     after element uml::Class created and inserted in uml::Package[packagedElement]
-    call createJavaClass(newValue)
+    call {
+        createOrFindJavaClass(newValue)
+        createRootJavaClass(newValue)
+    }
+}
 
+routine createRootJavaClass(uml::Classifier umlClassifier) {
+    match{
+        require absence of java::Class corresponding to umlClassifier
+        require absence of java::CompilationUnit corresponding to umlClassifier
+        require absence of java::Package corresponding to umlClassifier.eContainer
+    }
+    action {
+        call createJavaClass(umlClassifier) // if the direct container is the model
+    }
 }
 
-routine createJavaClass(uml::Classifier umlClassifier) {
-	match{
+routine createOrFindJavaClass(uml::Classifier umlClassifier) {
+    match{
         require absence of java::Class corresponding to umlClassifier
         require absence of java::CompilationUnit corresponding to umlClassifier
-	}
+        val javaPackage = retrieve java::Package corresponding to umlClassifier.eContainer
+    }
+    action {
+        call {
+            val foundClass = findClassifier(umlClassifier.name, javaPackage, org.emftext.language.java.classifiers.Class)
+            if(foundClass === null) {
+                createJavaClass(umlClassifier)
+            } else {
+                addMissingClassifierCorrespondence(umlClassifier, foundClass)
+            }
+        }
+    }
+}
+
```

Comment:
I added a special case (`createRouteJavaClass`) that covers the situation where a UML package is directly created in the model without a normal parent package. In this case, there is no corresponding Java package and the class needs to be created directly. This is also the case for the interface creation further down.